source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -176,20 +182,12 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
             throw mx::Exception("Unexpected DG node type.");
         }
 
-        std::string documentString = mx::writeToXmlString(materialXData->getDocument());
-
-        materialXNode->setMaterialXData(std::move(materialXData));
+        materialXNode->setData(documentFilePath, xmlElementPath, std::move(materialXData));
```

Comment:
Just one call now to pass all the necessary arguments from the command to the node. We don't want to set attribute values here because that would lead to unnecessary updates.